src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/26.diff.sql

```diff
@@ -0,0 +1,281 @@
+
+-- Deleting duplicate rows based on all columns
+EXEC dbo.LogSchemaMigrationProgress 'Deleting redundant rows from dbo.TokenText'
+GO
+;WITH CTE AS (
+    SELECT ResourceTypeId, ResourceSurrogateId, SearchParamId, Text, IsHistory, ROW_NUMBER() 
+    OVER (
+		PARTITION BY ResourceTypeId, ResourceSurrogateId, SearchParamId, Text, IsHistory
+		ORDER BY ResourceTypeId, ResourceSurrogateId, SearchParamId, Text
+	) row_num
+	FROM dbo.TokenText
+)
+DELETE FROM CTE WHERE row_num > 1
+GO
+
+-- Create nonclustered primary key on the set of non-nullable columns that makes it unique
+IF NOT EXISTS (
+    SELECT * 
+	FROM sys.key_constraints 
+	WHERE name='PK_TokenText' AND type='PK')
+BEGIN
+    EXEC dbo.LogSchemaMigrationProgress 'Adding PK_TokenText'
+	ALTER TABLE dbo.TokenText 
+	ADD CONSTRAINT PK_TokenText PRIMARY KEY NONCLUSTERED(ResourceTypeId, ResourceSurrogateId, SearchParamId, Text)
+    WITH (DATA_COMPRESSION = PAGE, ONLINE=ON)
+    ON PartitionScheme_ResourceTypeId(ResourceTypeId)
+END
+GO
+
+/******************************************************************
+  Table - dbo.DateTimeSearchParam table
+*******************************************************************/
+
+-- Deleting duplicate rows based on all columns
+EXEC dbo.LogSchemaMigrationProgress 'Deleting redundant rows from dbo.DateTimeSearchParam'
+GO
+;WITH CTE AS (
+    SELECT ResourceTypeId, ResourceSurrogateId, SearchParamId, StartDateTime, EndDateTime, IsLongerThanADay, IsHistory, IsMin, IsMax, ROW_NUMBER() 
+    OVER (
+		PARTITION BY ResourceTypeId, ResourceSurrogateId, SearchParamId, StartDateTime, EndDateTime, IsLongerThanADay, IsHistory, IsMin, IsMax
+		ORDER BY ResourceTypeId, ResourceSurrogateId, SearchParamId
+	) row_num
+	FROM dbo.DateTimeSearchParam
+)
+DELETE FROM CTE WHERE row_num > 1
+GO
+
+-- Create nonclustered primary key on the set of non-nullable columns that makes it unique
+IF NOT EXISTS (
+    SELECT * 
+	FROM sys.key_constraints 
+	WHERE name='PK_DateTimeSearchParam' AND type='PK')
+BEGIN
+    EXEC dbo.LogSchemaMigrationProgress 'Adding PK_DateTimeSearchParam'
+	ALTER TABLE dbo.DateTimeSearchParam 
+	ADD CONSTRAINT PK_DateTimeSearchParam PRIMARY KEY NONCLUSTERED(ResourceTypeId, ResourceSurrogateId, SearchParamId, StartDateTime, EndDateTime)
+	WITH (DATA_COMPRESSION = PAGE, ONLINE=ON)
+	ON PartitionScheme_ResourceTypeId(ResourceTypeId) 
+END
+GO
+
+/************************************************************************
+ Insert an entry with id 0 to dbo.QuantityCode table
+************************************************************************/
+EXEC dbo.LogSchemaMigrationProgress 'Insert an entry with id 0 to dbo.QuantityCode table'
+IF NOT EXISTS (SELECT 1 FROM dbo.QuantityCode WHERE QuantityCodeId = 0 AND value = '')
+BEGIN
+    SET IDENTITY_INSERT dbo.QuantityCode ON;
+
+    Insert INTO dbo.QuantityCode (QuantityCodeId, Value)
+    Values (0, '')
+
+    SET IDENTITY_INSERT dbo.QuantityCode OFF;
+END;
```

Comment:
nit: I see semicolon(;) after END at few places and not at few places. Should we add (;) after END to be consistent?